locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -2133,11 +2133,17 @@ protected NodeMetadata findNodeOrThrow(ConfigBag config) {
         }
         NodeMetadata node = Iterables.getOnlyElement(candidateNodes);
 
-        String pkd = LocationConfigUtils.getOsCredential(config).checkNoErrors().logAnyWarnings().getPrivateKeyData();
+        OsCredential osCredentials = LocationConfigUtils.getOsCredential(config).checkNoErrors().logAnyWarnings();
+        String pkd = osCredentials.getPrivateKeyData();
+        String password = osCredentials.getPassword();
         if (Strings.isNonBlank(pkd)) {
             LoginCredentials expectedCredentials = LoginCredentials.fromCredentials(new Credentials(user, pkd));
             //override credentials
             node = NodeMetadataBuilder.fromNodeMetadata(node).credentials(expectedCredentials).build();
+        } else if (Strings.isNonBlank(password)) {
+            LoginCredentials expectedCredentials = LoginCredentials.fromCredentials(new Credentials(user, password));
+            //override credentials
+            node = NodeMetadataBuilder.fromNodeMetadata(node).credentials(expectedCredentials).build();
```

Comment:
@andreaturli not sure I follow. The if-else means that it preserves previous behaviour when there is privateKeyData. If someone supplies both privateKeyData and a password, then the privateKeyData takes precedence. If I moved this out of the else/if, then it would overwrite the credentials set at lines 2138-2140.